silx/io/url.py

```diff
@@ -297,7 +298,10 @@ def slice_to_string(data_slice):
             if self.__data_path is not None:
                 queries.append("path=" + self.__data_path)
             if self.__data_slice is not None:
-                data_slice = ",".join([slice_to_string(s) for s in self.__data_slice])
+                if isinstance(self.__data_slice, Iterable):
+                    data_slice = ",".join([slice_to_string(s) for s in self.__data_slice])
+                else:
+                    data_slice = str(self.__data_slice)
```

Comment:
Probably best to use `slice_to_string` rather than `str`?

Reply:
Yes, totally miss that one thanks